### ISSUE TITLE
Portability: Assigning a pointer to an integer is not portable in sirhelpers.c:233:21 (branch: syslog)

### ISSUE THREAD

**Issue by johnsonjh** [aremmell/libsir#19]
from Cppcheck:

`sirhelpers.c:233:21: portability: Assigning a pointer to an integer is not portable [AssignmentAddressToInteger]`
```c
        errno_t ret = localtime_s(buf, timer);
```

**Comment by aremmell** [comment]
That one is wrapped in `#ifdef __WIN__` so I believe it's correct. Take a look, and close this if you feel it's correct.

**Comment by johnsonjh** [comment]
Is that portable on all Windows platforms (ARM and ARM64, etc)?  We do build for them ... I don't use Windows enough to know!  

Worst case, an explicit cast should silence the warning, if it's right.  I'll have to check further.

**Comment by aremmell** [comment]
> Is that portable on all Windows platforms (ARM and ARM64, etc)? We do build for them ... I don't use Windows enough to know!

Yeah, it should be standard across any architecture through the Win32 API: https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/localtime-s-localtime32-s-localtime64-s?view=msvc-170

**Comment by johnsonjh** [comment]
Fixed in syslog
